packages/flutter/lib/src/gestures/force_press.dart

```diff
@@ -0,0 +1,255 @@
+  // has just become greater than the ForcePressGestureRecognizer.startPressure.
+  started,
+
+  // A pointer is down and the pressure of the pointer has just become greater
+  // than the ForcePressGestureRecognizer.peakPressure. Even after a pointer
+  // crosses this threshold, onUpdate callbacks will still be sent.
+  peaked,
+}
+
+/// Details object for callbacks that use [GestureForcePressStartCallback],
+/// [GestureForcePressPeakCallback], [GestureForcePressEndCallback] or
+/// [GestureForcePressUpdateCallback].
+///
+/// See also:
+///
+///  * [ForcePressGestureRecognizer.onStart], [ForcePressGestureRecognizer.onPeak],
+///    [ForcePressGestureRecognizer.onEnd], and [ForcePressGestureRecognizer.onUpdate]
+///    which use [ForcePressDetails].
+///  * [ForcePressUpdateDetails], the details for [ForcePressUpdateCallback].
+class ForcePressDetails {
+  /// Creates details for a [GestureForcePressStartCallback],
+  /// [GestureForcePressPeakCallback] or [GestureForcePressEndCallback].
+  ///
+  /// The [globalPosition] argument must not be null.
+  ForcePressDetails({
+    @required this.globalPosition,
+    @required this.pressure,
+  }) : assert(globalPosition != null),
+       assert(pressure != null);
+
+  /// The global position at which the function was called.
+  final Offset globalPosition;
+
+  /// The pressure of the pointer on the screen.
+  final double pressure;
+}
+
+/// Signature used by a [ForcePressGestureRecognizer] for when a pointer has
+/// pressed with at least [ForcePressGestureRecognizer.startPressure].
+typedef GestureForcePressStartCallback = void Function(ForcePressDetails details);
+
+/// Signature used by [ForcePressGestureRecognizer] for when a pointer that has
+/// pressed with at least [ForcePressGestureRecognizer.peakPressure].
+typedef GestureForcePressPeakCallback = void Function(ForcePressDetails details);
+
+/// Signature used by [ForcePressGestureRecognizer] during the frames
+/// after the triggering of a [ForcePressGestureRecognizer.onStart] callback.
+typedef GestureForcePressUpdateCallback = void Function(ForcePressDetails details);
+
+/// Signature for when the pointer that previously triggered a
+/// [ForcePressGestureRecognizer.onStart] callback is no longer in contact
+/// with the screen.
+typedef GestureForcePressEndCallback = void Function(ForcePressDetails details);
+
+/// Recognizes a force press on devices that have force sensors.
+///
+/// Only the force from a single pointer is used to invoke events. A tap
+/// recognizer will win against this recognizer on pointer up as long as the
+/// pointer has not pressed with a force greater than
+/// [ForcePressGestureRecognizer.startPressure]. A long press recognizer will
+/// win when the press down time exceeds the threshold time as long as the
+/// pointer's pressure was never greater than
+/// [ForcePressGestureRecognizer.startPressure] in that duration.
+///
+/// As of 11-21-18 only iPhone 6S devices and higher numbered devices have force
```

Comment:
If you want to include a date please use the format "YYYY-MM-DD" or "Month YYYY", otherwise the date is wildly ambiguous.